src/main/java/at/hannibal2/skyhanni/features/mining/crystalhollows/CrystalHollowsLobbyAgeWarning.kt

```diff
@@ -0,0 +1,100 @@
+import kotlin.math.abs
+
+class CrystalHollowsLobbyAgeWarning {
+    private val config get() = SkyHanniMod.feature.mining.crystalHollowsLobbyAgeWarning
+    private val mc get() = Minecraft.getMinecraft()
+
+    private val THE_CRYSTAL_HOLLOWS = IslandType.CRYSTAL_HOLLOWS
+
+    private val TICKS_PER_MC_DAY: Long = 24000L
+
+    private val patternGroup = RepoPattern.group("mining.crystalhollows.lobbyage")
+    private val playerCountTabListPattern by patternGroup.pattern(
+        "playercount.tablist",
+        " *(?:§.)*Players[\\S ]*[({\\[](?<playerCount>[\\S ]+)[)}\\]]"
+    )
+
+    private var lobbyIsPastAgeThreshold: Boolean = false
+    private var oldPlayerCount: Long = 0L
+    private var oldLobbyAge: Long = 0L
+    private var minPlayers: Int = 4
+    private var minLobbyAgeMCDays: Int = 18
+    private var maxLobbyAgeMCDays: Int = 25
+
+    private fun getLobbyAgeInMinecraftDays(): Long = (mc.theWorld.worldTime) / TICKS_PER_MC_DAY //world day changes at 6AM in skyblock time
+
+    private fun processPlayerCount() {
+        for (line in TabListData.getTabList()) {
+            playerCountTabListPattern.matchMatcher(line) {
+                val playerCount = group("playerCount").formatNumber()
+                if (oldPlayerCount == playerCount) return
+                oldPlayerCount = playerCount
+                if (config.playerCountReminders) LorenzUtils.chat("§a$playerCount players are currently in this Crystal Hollows lobby.")
+                if (!lobbyIsPastAgeThreshold) return
+                if (playerCount in 1..minPlayers) LorenzUtils.chat("§cThere are $playerCount players remaining in this Crystal Hollows lobby. §4§lIt will shut down very soon.")
+                else LorenzUtils.chat("There are $playerCount players remaining in this Crystal Hollows lobby. §cIt will shut down when there are fewer than $minPlayers people left, or when this lobby reaches Day $maxLobbyAgeMCDays.")
+            }
+        }
+    }
+
+    private fun processLobbyAge() {
+        val lobbyAge = getLobbyAgeInMinecraftDays()
+        if (oldLobbyAge == lobbyAge) return
+        oldLobbyAge = lobbyAge
+        if (config.lobbyAgeReminders && !lobbyIsPastAgeThreshold) LorenzUtils.chat("§aThis Crystal Hollows lobby is currently at Day $lobbyAge, ${StringUtils.optionalPlural(number = abs(minLobbyAgeMCDays - lobbyAge).toInt(),"day", plural = "days")} away from Day $minLobbyAgeMCDays.")
+        if (lobbyAge in minLobbyAgeMCDays..< maxLobbyAgeMCDays && !lobbyIsPastAgeThreshold) {
+            lobbyIsPastAgeThreshold = true
+            LorenzUtils.chat("This Crystal Hollows lobby has reached Day $minLobbyAgeMCDays. It no longer accepts new players, §cand will shut down on Day $maxLobbyAgeMCDays or when there are fewer than $minPlayers people left.")
+        } else if (lobbyAge >= maxLobbyAgeMCDays) LorenzUtils.chat("§cThis Crystal Hollows lobby has reached Day $maxLobbyAgeMCDays. §4§lIt will shut down very soon.")
```

Comment:
This is unreadable also use this https://github.com/hannibal002/SkyHanni/pull/975